### ISSUE TITLE
IHT routine failing due to numpy call on torch tensor

### ISSUE THREAD

**Issue by pushkalkatara** [microsoft/EdgeML#119]
IHT routine fails in BonsaiTrainer due to use of np.copy() on Torch tensor.

```
Epoch Number: 32

Classification Train Loss: 0.1960460032439894
Training accuracy (Classification): 0.9822221928172641
Test accuracy 0.931241
MarginLoss + RegLoss: 0.18808786571025848 + 0.13880659639835358 = 0.32689446210861206


Epoch Number: 33
Traceback (most recent call last):
  File "bonsai_example.py", line 94, in <module>
    main()
  File "bonsai_example.py", line 89, in main
    dataDir, currDir)
  File "/home/pushkalkatara/pytorch/lib/python3.5/site-packages/edgeml-0.2.1-py3.5.egg/edgeml_pytorch/trainer/bonsaiTrainer.py", line 337, in train
  File "/home/pushkalkatara/pytorch/lib/python3.5/site-packages/edgeml-0.2.1-py3.5.egg/edgeml_pytorch/trainer/bonsaiTrainer.py", line 129, in runHardThrsd
  File "/home/pushkalkatara/pytorch/lib/python3.5/site-packages/numpy/lib/function_base.py", line 792, in copy
    return array(a, order=order, copy=True)
  File "/home/pushkalkatara/pytorch/lib/python3.5/site-packages/torch/tensor.py", line 458, in __array__
    return self.numpy()
RuntimeError: Can't call numpy() on Variable that requires grad. Use var.detach().numpy() instead.
```

**Comment by pushkalkatara** [comment]
https://github.com/microsoft/EdgeML/pull/120

**Comment by adityakusupati** [comment]
@SachinG007 I though you tested all the example scripts. Can you please explain this?

**Comment by pushkalkatara** [comment]
Solved and merged - #120 . Closing.
